File: cpp/Two Pointers/triplet_sum_brute_force.cpp
#include <algorithm> 
#include <set>       
#include <vector>

std::vector<std::vector<int>> tripletSumBruteForce(std::vector<int>& nums) {
    int n = nums.size();
    // Use a hash set to ensure we don't add duplicate triplets.
    std::set<std::vector<int>> triplets;
    // Iterate through the indexes of all triplets.
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            for (int k = j + 1; k < n; k++) {
                if (nums[i] + nums[j] + nums[k] == 0) {
                    // Sort the triplet before including it in the 
                    // hash set.
                    std::vector<int> triplet = {nums[i], nums[j], nums[k]};
                    std::sort(triplet.begin(), triplet.end());
                    triplets.insert(triplet);
                }
            }
        }
    }
    return {triplets.begin(), triplets.end()};
}

File: rust/Two Pointers/triplet_sum_brute_force.rs
use std::collections::HashSet;

fn triplet_sum_brute_force(nums: Vec<i32>) -> Vec<Vec<i32>> {
    let n = nums.len();
    // Use a hash set to ensure we don't add duplicate triplets.
    let mut triplets = HashSet::new();

    // Iterate through the indexes of all triplets.
    for i in 0..n {
        for j in (i + 1)..n {
            for k in (j + 1)..n {
                if nums[i] + nums[j] + nums[k] == 0 {
                    // Sort the triplet before including it in the hash set.
                    let mut triplet = vec![nums[i], nums[j], nums[k]];
                    triplet.sort();
                    triplets.insert(triplet);
                }
            }
        }
    }

    triplets.into_iter().collect()
}
